### PR TITLE
feat(release-brancher): add argument to override pull request title

### LINKED ISSUE #2568: release-brancher: allow customization of pull request to default branch
Allow the developer running the script to either customize the PR title and body and commit message.

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #2568

```
release-brancher create-pull-request \
  --branch-name="2.3.x" \
  --target-tag="v2.3.4" \
  --repo="<repo-owner>/<repo-name>" \
  --pull-request-title="feat: next release from mainline is 2.4.0"
  --release-type=java-backport
```

# My manual test

It worked

```
suztomo-macbookpro44% release-brancher create-pull-request --branch-name="1.40.0-sp.4" --target-tag="v1.40.0" --repo="suztomo/google-http-java-client"  --release-type=java-lts-on-patch --pull-request-title="feat: next release from mainline is 1.41.0"
creating branch 1.40.0-sp.4 as SHA f77608fbfb41410a8f62f40bc5a2df256c751d52
```

Screen Shot 2021-09-28 at 15 58 35

It created https://github.com/suztomo/google-http-java-client/pull/16 and https://github.com/suztomo/google-http-java-client/pull/15

---

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/repo-automation-bots/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #2568 🦕

### CODE REVIEW

File: packages/release-please/src/config-schema.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
                         "go-yoshi",
                         "java-bom",
                         "java-lts",
+                        "java-lts-no-sp",
```
Comment (suztomo) [inline packages/release-please/src/config-schema.json:26]: I couldn't find a better name.

File: packages/release-please/src/config-schema.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
                         "go-yoshi",
                         "java-bom",
                         "java-lts",
+                        "java-lts-on-patch",
```
Comment (suztomo) [inline packages/release-please/src/config-schema.json:26]: I think this is better than "java-lts-no-sp".

**Comment by suztomo** [comment]
The default branch name has become "main" from "master" in code-suggester.

https://github.com/googleapis/code-suggester/pull/262/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L144

**Comment by suztomo** [comment]
Now #2618 is merged.

File: packages/release-brancher/src/release-brancher.ts [hunk 3]
```diff
@@ -330,7 +336,7 @@ export class Runner {
         }
       }
     }
-    const message = 'feat: configure initial sp version';
+    const message = 'feat: configure the protected branch';
```
Comment (suztomo) [inline packages/release-brancher/src/release-brancher.ts:339]: @chingor13 Does "configure initial sp version" have special effect somewhere? In past (https://github.com/googleapis/repo-automation-bots/pull/1674/files#diff-4b2e8d3f62c327048db13c5f552d94303eac7ed241728c5fd5dbc3e5842961c9R327), you updated this message from 'ci: setup workflows for new release branch' to 'feat: configure initial sp version'.

File: packages/release-please/src/config-schema.json [hunk 4]
```diff
@@ -23,6 +23,7 @@
                         "go-yoshi",
                         "java-bom",
                         "java-lts",
+                        "java-backport",
```
Comment (suztomo) [inline packages/release-please/src/config-schema.json:26]: Now it's "java-backport"

**Comment by suztomo** [comment]
"ci / docs" failed but I believe it's irrelevant.

```
rewrite repo to suztomo/repo-automation-bots
Scanning **/*.md
Error: [429] https://www.github.com/googleapis/repo-automation-bots/commit/b6ed87e64d6c7599ab463337c3b0990e21ad7a13
Error: [429] https://www.github.com/googleapis/repo-automation-bots/commit/90430f3024cfd4c502f3001ab90ea585f8b8a85b
Error: [429] https://www.github.com/googleapis/repo-automation-bots/commit/4cb9d40c715a9a795e54711811e509df85174e60
Error: [429] https://www.github.com/googleapis/repo-automation-bots/commit/534e26bce3efb81556c7ce49bb84f2082deb1121
Error: [429] https://www.github.com/googleapis/repo-automation-bots/commit/a99805cacd2c534164c90c90072a1ab72fdccc58
Error: [429] https://www.github.com/googleapis/repo-automation-bots/commit/42e9ffc19ef8f0349b243969a7306129a27b4fa2
Error: [429] https://www.github.com/googleapis/repo-automation-bots/commit/ea93fe27d5401e9ca32e08f50914a03a8712a08b
Error: [429] https://github.com/googleapis/repo-automation-bots/commit/0032e84
Error: [429] https://www.github.com/googleapis/repo-automation-bots/commit/22c17c080235a0c1e93f05fb569004d648115bc0
Scanned total of 399 links!
Error: Detected 9 broken links.
 packages/gcf-utils/CHANGELOG.md
   [429] https://www.github.com/googleapis/repo-automation-bots/commit/b6ed87e64d6c7599ab463337c3b0990e21ad7a13
   [429] https://www.github.com/googleapis/repo-automation-bots/commit/90430f3024cfd4c502f3001ab90ea585f8b8a85b
   [429] https://www.github.com/googleapis/repo-automation-bots/commit/4cb9d40c715a9a795e54711811e509df85174e60
   [429] https://www.github.com/googleapis/repo-automation-bots/commit/534e26bce3efb81556c7ce49bb84f2082deb1121
   [429] https://www.github.com/googleapis/repo-automation-bots/commit/a99805cacd2c534164c90c90072a1ab72fdccc58
   [429] https://www.github.com/googleapis/repo-automation-bots/commit/42e9ffc19ef8f0349b243969a7306129a27b4fa2
   [429] https://www.github.com/googleapis/repo-automation-bots/commit/ea93fe27d5401e9ca32e08f50914a03a8712a08b
   [429] https://github.com/googleapis/repo-automation-bots/commit/0032e84
 packages/label-utils/CHANGELOG.md
   [429] https://www.github.com/googleapis/repo-automation-bots/commit/22c17c080235a0c1e93f05fb569004d648115bc0
```

File: packages/release-brancher/src/release-brancher.ts (hunk 3 above)
Comment (chingor13) [reply packages/release-brancher/src/release-brancher.ts:339]: In this case, any releasable commit (`feat`, `fix`, etc, will trigger a new library release). For the -sp versions, we wanted to force a release. For the new backport releases, we don't need to force a new release, so `ci` is probably a better message.

**Review by chingor13 — APPROVED** [review]

File: packages/release-brancher/src/release-brancher.ts (hunk 3 above)
Comment (suztomo) [reply packages/release-brancher/src/release-brancher.ts:339]: Created https://github.com/googleapis/repo-automation-bots/pull/2622#pullrequestreview-766165606

**Pull request merged**